### PR TITLE
iOS: Track user id.

### PR DESCRIPTION
**Pull request by pablorueda**
## Description
In conversations between Udi, Gavri and Aviran, has been aproved that implementation team creates a PR for a new functionality to the google analytic plugin.

This new functionality is the ability to the analytics plugin to track the user_id: https://firebase.google.com/docs/analytics/userid

We want this functionality to be public available for all apps, but difficult to configure, as we don't want people to configure if they don't know what they are doing. For that reason it should only be enable if a new parameter in plugin configurations named called user_id has the value should allow-tracking-user-id-for-app-family-<APP_FAMILY_ID>

The identifier for the user_id will be setup in those login plugins that desire to implement this, under the local storage, with key "user_id" under the "login" namespace.

https://applicaster.atlassian.net/browse/IM-501

## Checklist

* [x] PR is scoped to one task
* [x] Tests are included
* [x] Documentation is included

* [x] This PR is not making any UI change
* [ ] This PR is making a UI change
  * [ ] Screenshot(s) included

## QA :

* required test cases:
The only way to test this is end to end, you need access to TVA Firebase console. You should check that analytics are being tracked using the user id, for that just launch the app and login. After login you should see that your tracking contain the user id.
* specific apps / plugins to test :
Television Academy

### CODE REVIEW

**Review by kononenkoAnton — APPROVED** [review]

File: Manifests/ios.json [hunk 1]
```diff
@@ -31,6 +31,12 @@
       "label": "Send User Data",
       "default": 1,
       "tooltip_text": "Sends any user data collected in the app. Per Google&apos;s policies, we will not send any user data identified as PII (Personally Identifiable Information)."
+    },
+    {
+      "type": "text",
+      "key": "user_id",
```
Comment (snooky23) [inline Manifests/ios.json:37]: Should be app family id also in the type, label and tooltip

File: iOS/ZappAnalyticsPluginFirebase/APAnalyticsProviderFirebase.swift [hunk 2]
```diff
@@ -131,6 +141,14 @@ open class APAnalyticsProviderFirebase: ZPAnalyticsProvider {
                 }
                 Analytics.setUserProperty(value, forName: key)
             }
+            
+            //We want to add the user id only if it is configured as allow-tracking-user-id-for-app-family-<APP_FAMILY_ID> in the plugin configurations
+            if let userIDParameter = configurationJSON?[UserIDConstants.userIDJSONConfig] as? String,
```
Comment (snooky23) [inline iOS/ZappAnalyticsPluginFirebase/APAnalyticsProviderFirebase.swift:146]: Put this logic in a fileprivate method and call this here.
Add error logs.
Use guard statement and separate this logic.

**Review by snooky23 — CHANGES_REQUESTED** [review]
Add this info to the plugin readme file. If we are not using one then please create one and write only this info there.

File: Manifests/ios.json (hunk 1 above)
Comment (PRuedaApplicaster) [reply Manifests/ios.json:37]: No. This was secified by product. We want it to be difficult for a Zapper to configure, as to configure has legal implies. So we don't want to document anything or change change the key/tool tip.

File: iOS/ZappAnalyticsPluginFirebase/APAnalyticsProviderFirebase.swift (hunk 2 above)
Comment (PRuedaApplicaster) [reply iOS/ZappAnalyticsPluginFirebase/APAnalyticsProviderFirebase.swift:146]: To abstract the logic to a method is fine, but I don't understand what you want to check in an error log there. I don't think we should track any error if this logic is not executing.

Comment (PRuedaApplicaster) [reply iOS/ZappAnalyticsPluginFirebase/APAnalyticsProviderFirebase.swift]: Commited a new change with refactor of the method.

**Review by snooky23 — APPROVED** [review]

**Review by lomudi — APPROVED** [review]
💯

**Pull request merged**